distributed/tests/test_client.py

```diff
@@ -6515,3 +6515,30 @@ async def test_workers_collection_restriction(c, s, a, b):
     future = c.compute(da.arange(10), workers=a.address)
     await future
     assert a.data and not b.data
+
+
+@gen_cluster(client=True)
+async def test_custom_key_with_batches(c, s, a, b):
+    """ Test of <https://github.com/dask/distributed/issues/4588>"""
+
+    futs = c.map(
+        lambda x: x ** 2,
+        list(range(10)),
```

Comment:
```suggestion
        range(10),
```

Reply:
This one won't work, right?

Reply:
Interesting, I thought that `range` was a generator, so wouldn't work with `client.map` but it works.

Reply:
`client.map` should work with anything that's an iterator